xapian-core/cluster/kmeans.cc

```diff
@@ -72,7 +79,7 @@ void
 KMeans::initialise_points(const MSet &source)
 {
     LOGCALL_VOID(API, "KMeans::initialise_points", source);
-    TermListGroup tlg(source);
+    tlg.construct_tlg(source);
```

Comment:
This prolongs the lifetime of the `TermListGroup` and the (potentially large) `unordered_map` it contains.

I'd be inclined to keep `tlg` as a local here, and instead make `stopper` a member of `KMeans` which gets passed as an optional parameter to the `TermListGroup` constructor.

That also avoids having to worry about the semantics of calling `construct_tlg()` more than once on the same object.